libs/components/auto-scale/auto-scale-img.directive.ts

```diff
@@ -0,0 +1,24 @@
+import { ChangeDetectorRef, Directive, ElementRef, HostListener, Self } from '@angular/core';
+
+@Directive({
+  selector: '[gewdAutoScaleImg]'
+})
+export class AutoScaleImgDirective {
+
+  @HostListener('load')
+  public onLoad() {
+    this._cd.markForCheck();
+  }
+
+  constructor(
+    @Self() public elemRef: ElementRef,
```

Comment:
is the `Self` needed? I used some other directives and the `@Self` was never needed

Reply:
Think of `self` as stricter than if there was no self. This `self` means, that you can only add this directive to a `img` Tag. If it wasn't there Angular would start looking downwards in the DOM for the next ElementRef. I don't think you could add a directive and not have an `ElementRef`, but I think it shows, that the focus is the HTML Element where you put the directive on